hipcheck/src/policy_exprs/token.rs

```diff
@@ -70,6 +73,20 @@ fn lex_ident(input: &mut Lexer<'_, Token>) -> Result<String> {
 	Ok(input.slice().to_owned())
 }
 
+/// Lex a JSON Pointer.
+fn lex_json_pointer(input: &mut Lexer<'_, Token>) -> Result<String> {
+	// Remove the initial '$' character.
+	// Slice safety: the slice is guaranteed to be at least 1 character long,
+	// because the regex for JSONPointer contains a mandatory '$' character.
+	let s = &input.slice()[1..];
+	if let Some(chr) = s.chars().next() {
+		if chr != '/' {
+			return Err(LexingError::InvalidJSONPointer(s.to_owned()));
+		}
```

Comment:
`InvalidJSONPointer` is rather vague for a scenario where we know the problem was a missing `/`. Could we either add another enum as a field in `InvalidJSONPointer`, or ensure that the `Display` impl for the this variant explicitly mentions missing a `/` if that's all it will be used for? Being told it's invalid but not explaining why when we have the power to would be frustrating for users.

Reply:
You're right, the error variant name is vague for how specific this invalid syntax situation is.

I'd be happy to just rename this error, and we can add more variants if we find other ways that JSON Pointers can be invalid in syntax.

I did mention the underlying problem more specifically in the display text for the error, but it feels like it could be a footgun for later modification to this code, since there's nothing enforcing that we only use this error type for a missing '/' character.

Reply:
I've renamed the error variant to `JSONPointerMissingInitialSlash` in d2a820f. Verbose, but specific.